File: backend-dev/src/entities/user.entity.ts
import { Column, CreateDateColumn, Entity, PrimaryColumn, UpdateDateColumn } from "typeorm";

@Entity("User")
export class UserEntity {
  @PrimaryColumn({ name: "UserId" })
  userId: string;

  @Column({ name: "Email", unique: true })
  email: string;

  @Column({ name: "Uid", unique: true })
  uid: string;

  @Column({ name: "DisplayName" })
  displayName: string;

  @Column({ name: "AvatarId" })
  avatarId: string;

  @Column({ name: "Message" })
  message: string;

  @CreateDateColumn({ name: "CreatedAt", type: "timestamptz" })
  createdAt: Date;

  @UpdateDateColumn({ name: "UpdatedAt", type: "timestamptz", nullable: true })
  updatedAt: Date;
}

File: apps/backend/src/main/java/app/sandori/arena/api/domain/user/UserEntity.java
package app.sandori.arena.api.domain.user;

import app.sandori.arena.api.global.util.IdGenerator;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.LocalDateTime;

@Entity
@Table(name = "users")
public class UserEntity {

    @Id
    @Column(name = "userId")
    private String userId;

    @Column(name = "uid", nullable = false, unique = true)
    private String uid;

    @Column(name = "email", nullable = false, length = 255)
    private String email;

    @Column(name = "createdAt", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updatedAt", nullable = false)
    private LocalDateTime updatedAt;

    @Column(name = "deletedAt")
    private LocalDateTime deletedAt;

    protected UserEntity() {
    }

    public UserEntity(String uid, String email) {
        this.userId = IdGenerator.generate();
        this.uid = uid;
        this.email = email;
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }

    public String getUserId() { return userId; }
    public String getUid() { return uid; }
    public String getEmail() { return email; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public LocalDateTime getDeletedAt() { return deletedAt; }
}
